.github/actions/build-keycloak/action.yml

```diff
@@ -43,13 +43,37 @@ runs:
       name: NPM cache
       uses: ./.github/actions/npm-cache
 
+    # Remove once https://github.com/keycloak/keycloak/issues/19299 is solved
+    ########################################################################################################
+    - id: check-adapter-changes
+      if: github.event_name == 'pull_request'
+      name: Check changes for WildFly adapters
+      shell: bash
+      # If there are no changes for WildFly adapters, we use adapters built in the latest nightly build
+      run: |
+        WF_ADAPTERS_REGEX="^adapters/oidc/wildfly|^adapters/saml/wildfly"
+        
+        git fetch origin --tags --force
+        
+        echo "GIT_WF_ADAPTERS_DIFF=$(git diff origin/main --name-only | egrep -ic -e "$WF_ADAPTERS_REGEX")" >> $GITHUB_ENV
+        echo "NIGHTLY_DIFF=$(git diff nightly --name-only | egrep -ic -e "$WF_ADAPTERS_REGEX")" >> $GITHUB_ENV
+
+    - id: set-maven-profile
+      if: ${{ github.event_name != 'pull_request' || env.GIT_WF_ADAPTERS_DIFF != 0 || env.NIGHTLY_DIFF != 0}}
+      name: Set profile for building WildFly adapters
+      shell: bash
+      run: |
+        echo "MVN_PROFILES=-Pbuild-wildfly-adapters" >> $GITHUB_ENV
+        echo "WildFly adapters will be built in our codebase"
+    ########################################################################################################
```

Comment:
@stianst Could you please check it, as you've been working on the GHAs? Thank you